### PR TITLE
Fix LaTeX equations for KaTeX

### LINKED ISSUE #11358: Latex equation broken in RZZGate documentation
### Environment

Online docs

### What is happening?

On the RZZGate docs page: https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.library.RZZGate

An equation that is a helpful reference for getting factors of 2 correct does not render. Mousing over gives the tooltip shown with the error message suggesting to use renewcommand instead of newcommand:
image

### How can we reproduce the issue?

https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.library.RZZGate

### What should happen?

Equation should render without an error

### Any suggestions?

May just need to replace \newcommand with \renewcommand

### PR DESCRIPTION
**Pull request by kevinsung**
### Summary
`\th` is already a latex command, so `\newcommand{\th}` raises an error in our public docs, which uses KaTeX. It seems that it renders in the Sphinx docs because it uses MathJax, which is lenient. See https://github.com/Qiskit/documentation/issues/59#issuecomment-1847804047.

### CODE REVIEW

**Comment by jakelishman** [comment]
It's good to get these fixed - this fixes #11358 open on this repo. If you wanted to remove the macros entirely and just inline the `\frac\theta2` where appropriate, I'd be in favour of that, though I'm also fine if you want to continue with the macro rename you've got in progress.

**Comment by kevinsung** [comment]
Let's just keep the macros for now.

File: docs/conf.py [hunk 1]
```diff
@@ -40,11 +40,11 @@
     "sphinx.ext.napoleon",
     "sphinx.ext.autodoc",
     "sphinx.ext.autosummary",
-    "sphinx.ext.mathjax",
     "sphinx.ext.intersphinx",
     "sphinx.ext.doctest",
     "matplotlib.sphinxext.plot_directive",
     "reno.sphinxext",
+    "sphinxcontrib.katex",
```
Comment (jakelishman) [inline docs/conf.py:47]: This is a good idea, since the separate docs build uses KaTeX now - best to keep us a bit more in sync.

**Review by jakelishman — APPROVED** [review]
Thanks for this - good to get these things sorted, and it's a good idea to swap to KaTeX as our CI rendering engine as well, so the docs artefacts can be checked.

File: docs/conf.py (hunk 1 above)
Comment (kevinsung) [reply docs/conf.py:47]: Oh, I actually was just testing it out and included here by accident ... I was going to do it in a separate PR. But it worked fine in my tests.

**Pull request merged**